pkg/reloader/reloader.go

```diff
@@ -207,7 +207,21 @@ func (r *Reloader) apply(ctx context.Context) error {
 
 	h := sha256.New()
 	for _, ruleDir := range r.ruleDirs {
-		err := filepath.Walk(ruleDir, func(path string, f os.FileInfo, err error) error {
+		walkDir := ruleDir
+		// check if ruleDir is a symlink
```

Comment:
This code has the classical TOCTTOU problem - time-of-checking-till-time-of-use. Obviously, it might not make a difference here since we aren't doing anything like `setuid(2)` later. Either way, I'd recommend to use https://golang.org/pkg/path/filepath/#EvalSymlinks here - plus the code will be simplified.

Reply:
good catch. wasn't aware of the `EvalSymlinks` func, and the cleanliness alone makes it worth using, let alone resolving the TOCTTOU issue.